compiler/rustc_codegen_ssa/src/traits/builder.rs

```diff
@@ -272,6 +272,7 @@ pub trait BuilderMethods<'a, 'tcx>:
 
     // These are used by everyone except msvc
     fn cleanup_landing_pad(&mut self, pers_fn: Self::Value) -> (Self::Value, Self::Value);
+    fn filter_landing_pad(&mut self, pers_fn: Self::Value) -> (Self::Value, Self::Value);
```

Comment:
What is the difference between cleanup and filter? How should a backend implement it?

Reply:
It's purely metadata in LSDA. https://itanium-cxx-abi.github.io/cxx-abi/exceptions.pdf

A negative value in action record means filter, a positive value means catch, and zero (or absence of action record) means cleanup.

Reply:
This corresponds to section 7.5 "Exception Specification" of that document?

Reply:
Yeah. I call it `filter` because that's how LLVM calls it.

The code here doesn't actually specify any exceptions (corresponds to C++ `throw()`), so it's essentially catch-all. But it provides a way for the personality function to distinguish terminate vs cleanup.